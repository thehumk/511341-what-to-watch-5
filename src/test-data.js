export const noop = () => {};

export const MATCH = {params: {id: `1`}};

export const mockComments = [
  {
    comment: `123`,
    date: `2020-10-29T13:38:44.769Z`,
    id: 1,
    rating: 4.6,
    user: {
      id: 12,
      name: `Isaac`
    },
  },
  {
    comment: `1234`,
    date: `2020-10-29T13:38:45.769Z`,
    id: 2,
    rating: 4.5,
    user: {
      id: 13,
      name: `Isa`
    },
  }
];

/* eslint-disable */

export const mockFilms = [
  {
    background_color: `#73B39A`,
    background_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    id: 1,
    is_favorite: true,
    name: `Bronson`,
    poster_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    run_time: 92,
    scores_count: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    video_link: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
  {
    background_color: `#73B39A`,
    background_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    description: `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    director: `Nicolas Winding Refn`,
    genre: `Drama`,
    id: 2,
    is_favorite: false,
    name: `Bronson`,
    poster_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    preview_image: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    preview_video_link: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: 3.6,
    released: 2008,
    run_time: 92,
    scores_count: 109661,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    video_link: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  },
];

/* eslint-enable */

export const state = {
  DATA: {
    films: mockFilms,
    promoFilm: mockFilms[0],
    genresList: [`All`, `Action`, `Drama`],
    isApplicationReady: false,
  },
  FILMS_STATUS: {
    activeGenre: `All`,
    renderFilmsCount: 8,
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
    user: null,
  },
  COMMENTS: {
    comments: mockComments,
  },
};
