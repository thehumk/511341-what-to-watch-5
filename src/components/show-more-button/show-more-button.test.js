import React from 'react';
import renderer from 'react-test-renderer';
import ShowMoreButton from './show-more-button';
import {noop} from '../../test-data';

it(`Should ShowMoreButton render correctly`, () => {
  const tree = renderer.create(
      <ShowMoreButton clickHandler={noop}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
