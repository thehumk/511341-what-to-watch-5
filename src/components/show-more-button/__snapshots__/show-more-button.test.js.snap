// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ShowMoreButton render correctly 1`] = `
<button
  className="catalog__button"
  onClick={[Function]}
  type="button"
>
  Show more
</button>
`;
